### PR TITLE
Add toString to anonymous functions #531.

### PR DESCRIPTION
**Pull request by andrewsmartin**

### CODE REVIEW

**Comment by nevillelyh** [comment]
Threre're a couple more `Function*` utilities in https://github.com/spotify/scio/tree/master/scio-core/src/main/scala/com/spotify/scio/util
Should we cover them also?

**Comment by andrewsmartin** [comment]
Sure.

File: scio-core/src/main/scala/com/spotify/scio/util/Functions.scala [hunk 1]
```diff
@@ -195,4 +195,14 @@ private[scio] object Functions {
     override def reduceOption(accumulator: Iterable[T]): Option[T] = _mon.sumOption(accumulator)
   }
 
+  /** Helper trait to decorate anonymous functions with meaningful toString. */
+  private trait PrettyFn {
+    val callsite = CallSites.getCurrent
+    override def toString: String = callsite
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/util/Functions.scala]: Make it something like `s"anonymous function $callsite"`?

File: scio-core/src/main/scala/com/spotify/scio/util/Functions.scala [hunk 2]
```diff
@@ -195,4 +195,14 @@ private[scio] object Functions {
     override def reduceOption(accumulator: Iterable[T]): Option[T] = _mon.sumOption(accumulator)
   }
 
+  /** Helper trait to decorate anonymous functions with meaningful toString. */
```
Comment (andrewsmartin) [inline scio-core/src/main/scala/com/spotify/scio/util/Functions.scala]: @nevillelyh Small style question, would you prefer these stay here or move to another file `PrettyFn.scala`? I sort of prefer moving them so they're more visible now that they're being used across the package.

Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/util/Functions.scala]: Just this 1 trait right? Don't have a preference but I won't move it :)

Comment (andrewsmartin) [inline scio-core/src/main/scala/com/spotify/scio/util/Functions.scala]: Well the trait and the 3 lines below as well

**Comment by nevillelyh** [comment]
LGTM merge when tests pass.

**Pull request merged**